src/braket/circuits/circuit.py

```diff
@@ -643,6 +662,89 @@ def add_verbatim_box(
             self._has_compiler_directives = True
         return self
 
+    def _add_measure(self, target_qubits: QubitSetInput) -> None:
+        """Adds a measure instruction to the the circuit
+
+        Args:
+            target_qubits (QubitSetInput): target qubits to measure.
+        """
+        for idx, target in enumerate(target_qubits):
+            num_qubits_measured = (
+                len(self._measure_targets)
+                if self._measure_targets and len(target_qubits) == 1
+                else 0
+            )
+            self.add_instruction(
+                Instruction(
+                    operator=Measure(index=idx + num_qubits_measured),
+                    target=target,
+                )
+            )
+            if self._measure_targets:
+                self._measure_targets.append(target)
+            else:
+                self._measure_targets = [target]
+
+    def measure(self, target_qubits: QubitSetInput | None = None) -> Circuit:
+        """
+        Add a `measure` operator to `self` ensuring only the target qubits are measured.
+
+        Args:
+            target_qubits (QubitSetInput | None): target qubits to measure.
+                Default=None
+
+        Returns:
+            Circuit: self
+
+        Raises:
+            IndexError: If `self` has no qubits.
+            IndexError: If target qubits are not within the range of the current circuit.
+            ValueError: If the current circuit contains any result types.
+            ValueError: If the target qubit is already measured.
+
+        Examples:
+            >>> circ = Circuit.h(0).cnot(0, 1).measure([0])
+            >>> circ.print(list(circ.instructions))
+            [Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(0)]),
+            Instruction('operator': CNot('qubit_count': 2), 'target': QubitSet([Qubit(0),
+                Qubit(1)]),
+            Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(2)]),
+            Instruction('operator': Measure, 'target': QubitSet([Qubit(0)])]
+        """
+        # check whether measuring an empty circuit
+        if not self.qubits:
+            raise IndexError("Cannot measure an empty circuit.")
+
+        if isinstance(target_qubits, int):
+            target_qubits = [target_qubits]
+
+        # Check that the target qubits are on the circuit
+        if target_qubits and not all(qubit in self.qubits for qubit in target_qubits):
+            raise IndexError("Target qubits must be within the range of the current circuit.")
```

Comment:
Ambiguous whether this should be the behavior or just add the qubit to the circuit as with all the other operations.

Reply:
That's a fair point. Are there any specific use cases where a customer would want to measure a qubit not on the circuit?

Reply:
Yeah, it's possible for example that a user just wants to test readout error of the device. In which case they may run some gates on a small subset of qubits (or just add a time delay, without running any gates at all) and then measure all of the qubits on the device.

Reply:
In that case, this error is probably not needed as it prevents a use case. I can remove it.

Reply:
Makes sense, and having a unit test for that case would also be helpful if there isn't one already

Reply:
Added a unit test for `Circuit().h(0).cnot(0, 1).measure(4)`